### PR TITLE
Mike without redux

### PR DESCRIPTION
**Pull request by mikemum**
This PR mainly includes:
- merging the latest master with a merge conflict resolved 
- implementation attendance override by the admin
- implementation of user balance update upon creating an attendance
- implementation for the corresponding UI updates

### CODE REVIEW

File: src/components/attendanceRecord/sharedComponents/circleImg.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
 import React from 'react';
 
-const CircleImgWithName = (props) => {
-    let user =  props.user;
-    let name = user.firstname + ' ' + user.lastname;
+const CircleImg = (props) => {
+    let user = props.user;
+    user.imgUrl = 'https://cdn.iconscout.com/icon/free/png-512/avatar-370-456322.png';
```
Comment (sophiablitz) [inline src/components/attendanceRecord/sharedComponents/circleImg.js:5]: Why are you hardcoding the URL? Can we use real URL's?

File: src/components/attendanceRecord/sharedComponents/danceClassCard.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+const danceClassCard = (props) => {
+    return (
+        <React.Fragment>
+            <div className="dance-class-info">
+                <div>
+                    <img src={props.imgUrl} alt="salsa dance" />
```
Comment (sophiablitz) [inline src/components/attendanceRecord/sharedComponents/danceClassCard.js:8]: Can the alt text be dynamic to match the actual class?

File: src/components/attendanceRecord/sharedComponents/danceClassCard.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                    <h3>{props.title}</h3>
+                    <p>{props.description.substr(0, 140)}...</p>
+                </div>
+            </div>
+
+            <div className="dance-class-action">
+                <button
+                    className="btn btn-primary"
+                    onClick={() => this.onLearnMoreBtnClick()}
+                > LEARN MORE >></button>
```
Comment (sophiablitz) [inline src/components/attendanceRecord/sharedComponents/danceClassCard.js:20]: Can this be TItle Case? "Learn More"

File: src/components/attendanceRecord/sharedComponents/circleImg.js (hunk 1 above)
Comment (mikemum) [reply src/components/attendanceRecord/sharedComponents/circleImg.js:5]: replaced by dynamic url

File: src/components/attendanceRecord/sharedComponents/danceClassCard.js (hunk 2 above)
Comment (mikemum) [reply src/components/attendanceRecord/sharedComponents/danceClassCard.js:8]: replaced the place holder with the actual class name

**Pull request merged**